torch/distributions/binomial.py

```diff
@@ -64,25 +69,30 @@ def logits(self):
     def probs(self):
         return logits_to_probs(self.logits, is_binary=True)
 
+    @property
+    def param_shape(self):
+        return self._param.size()
+
     def sample(self, sample_shape=torch.Size()):
         shape = self._extended_shape(sample_shape) + (self.total_count,)
         return torch.bernoulli(self.probs.unsqueeze(-1).expand(shape)).sum(dim=-1)
 
     def log_prob(self, value):
         self._validate_log_prob_arg(value)
-        probs = clamp_probs(self.probs)
         log_factorial_n = math.lgamma(self.total_count + 1)
         log_factorial_k = torch.lgamma(value + 1)
         log_factorial_nmk = torch.lgamma(self.total_count - value + 1)
+        max_val = (-self.logits).clamp(min=0.0)
         return (log_factorial_n - log_factorial_k - log_factorial_nmk +
-                value * self.logits + self.total_count * torch.log1p(-probs))
+                value * self.logits + self.total_count * max_val -

```

Comment:
@alicanb - I wanted to eliminate the use of probs. Does this look fine to you?

Reply:
Sure, we can get rid of `clamp_probs` then, I think this was the only place that it's used